browser/ui/views/brave_javascript_tab_modal_dialog_view_views.cc

```diff
@@ -0,0 +1,172 @@
+          parent_web_contents,
+          alerting_web_contents,
+          title,
+          dialog_type,
+          message_text,
+          default_prompt_text,
+          std::move(dialog_callback),
+          std::move(dialog_force_closed_callback)),
+      alerting_web_contents_(alerting_web_contents) {
+  // JavaScriptTabModalDialogViewViews already created Widget.
+  auto* widget = GetWidget();
+  CHECK(widget);
+
+  widget->SetZOrderSublevel(kSublevelSecurity);
+
+  widget->widget_delegate()->set_desired_position_delegate(base::BindRepeating(
+      [](base::WeakPtr<BraveJavaScriptTabModalDialogViewViews> dialog_view) {
+        if (!dialog_view) {
+          return gfx::Point();
+        }
+        return dialog_view->GetDesiredPositionConsideringSplitView();
+      },
+      weak_ptr_factory_.GetWeakPtr()));
+
+  UpdateWidgetBounds();
+}
+
+BraveJavaScriptTabModalDialogViewViews::
+    ~BraveJavaScriptTabModalDialogViewViews() = default;
+
+base::WeakPtr<javascript_dialogs::TabModalDialogView>
+JavaScriptTabModalDialogManagerDelegateDesktop::CreateNewDialog(
+    content::WebContents* alerting_web_contents,
+    const std::u16string& title,
+    content::JavaScriptDialogType dialog_type,
+    const std::u16string& message_text,
+    const std::u16string& default_prompt_text,
+    content::JavaScriptDialogManager::DialogClosedCallback dialog_callback,
+    base::OnceClosure dialog_force_closed_callback) {
+  auto* browser = chrome::FindBrowserWithTab(alerting_web_contents);
+  if (!browser) {
+    // Can be popup up or other type of window.
+    return CreateNewDialog_ChromiumImpl(
+        alerting_web_contents, title, dialog_type, message_text,
+        default_prompt_text, std::move(dialog_callback),
+        std::move(dialog_force_closed_callback));
+  }
+
+  if (!SplitViewBrowserData::FromBrowser(browser)) {
+    // Split view isn't enabled.
+    return CreateNewDialog_ChromiumImpl(
+        alerting_web_contents, title, dialog_type, message_text,
+        default_prompt_text, std::move(dialog_callback),
+        std::move(dialog_force_closed_callback));
+  }
+
+  return (new BraveJavaScriptTabModalDialogViewViews(
+              web_contents_, alerting_web_contents, title, dialog_type,
+              message_text, default_prompt_text, std::move(dialog_callback),
+              std::move(dialog_force_closed_callback)))
+      ->weak_ptr_factory_.GetWeakPtr();
+}
+
+web_modal::WebContentsModalDialogHost*
+BraveJavaScriptTabModalDialogViewViews::GetModalDialogHost() {
+  web_modal::WebContentsModalDialogManager* manager =
+      web_modal::WebContentsModalDialogManager::FromWebContents(
+          alerting_web_contents_);
+  CHECK(manager);
+
+  auto* modal_dialog_host =
+      manager->delegate()->GetWebContentsModalDialogHost();
+  CHECK(modal_dialog_host);
+
+  return modal_dialog_host;
+}
+
+void BraveJavaScriptTabModalDialogViewViews::UpdateWidgetBounds() {
+  auto* widget = GetWidget();
+  if (!widget) {
```

Comment:
is there any reason in particular why this function tests for `!widget`, rather than `CHECK`ing it like the other places?